File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {
        Stack<Character> stack = new Stack<>();
        
        for(char ch:s.toCharArray()){
            // System.out.println(stack);
            if(stack.isEmpty() || stack.peek() != ch){
                stack.add(ch);
            } else{
                if(ch == stack.peek()) stack.pop();
            }
        }
        
        // System.out.println(stack);
        
        return stack.isEmpty() ? 1 : 0;
    }
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stack1 = list(s)[::-1]
    stack2 = []
    
    while stack1:        
        if stack1 and stack2 and stack1[-1] == stack2[-1]:
            stack1.pop()
            stack2.pop()
            continue
        
        stack2.append(stack1.pop())
        if stack1 and stack1[-1] == stack2[-1]:
            stack1.pop()
            stack2.pop()
        # print(stack1, stack2)
    
    if not stack1 and not stack2:
        return 1
        
    return 0